### PR TITLE
Scatter hover fix

### LINKED ISSUE #66: Scatter Chart mouseover
could potentially solve the match by color issue through: https://github.com/EQWorks/chart-system/pull/62#discussion_r449634840

or using the idea that the circles seem to be rendered in the same order of the data

### PR DESCRIPTION
**Pull request by geoerika**
Fix for hover, targeting data id instead of fill colour

Closes #66

### CODE REVIEW

File: src/components/scatter-chart/events.js [hunk 1]
```diff
@@ -2,36 +2,34 @@ import { DATA_HOVER_OPACITY } from '../../shared/constants/dimensions'
...
-      }
+export const onMouseEnter = (event, node) => {
+  const parentNode = node.target.parentNode.parentNode
+  let nodeId = node.target.id
+  // isolate circles from data sets that are not hovered and set their opacity
+  Array.from(parentNode.children)
+    .filter(ele =>
+      ele.children[0].tagName === 'circle')
+    .filter(ele => (
+      ele.children[0].getAttributeNode('id').value !== nodeId ))
```
Comment (Goondrious) [inline src/components/scatter-chart/events.js:15]: _pattern:_ these can be within one filter

File: src/components/scatter-chart/events.js [hunk 2]
```diff
@@ -2,36 +2,34 @@ import { DATA_HOVER_OPACITY } from '../../shared/constants/dimensions'
...
-  const parentNode = event.target.parentNode
-  const dataPoints = parentNode.getElementsByTagName('circle')
-  dataPoints.forEach(point => point.style.opacity = 1)
+export const onMouseLeave = (event, node) => {
+  const parentNode = node.target.parentNode.parentNode
+  Array.from(parentNode.children)
+    .filter(ele =>
+      ele.children[0].tagName === 'circle')
+    .forEach(ele => {
+      ele.children[0].style.opacity = 1
```
Comment (Goondrious) [inline src/components/scatter-chart/events.js:33]: _pattern:_ if we don't need to keep the array (e.g. `const dots = parentNode.children.filter`), then we could do both of these operations within the `.forEach` to save a loop (same for above)

File: src/shared/utils/index.js [hunk 3]
```diff
@@ -159,13 +160,15 @@ const setChartMargin = (
       legendTranslate = LEGEND_TRANSLATE_X
       const expandingLabelContainer = width - WIDTH_BREAKPOINT_3 - LEGEND_COLUMN_FIXED_ELEMENTS_WIDTH - legendTranslate
       legendLabelContainerWidth = Math.max(expandingLabelContainer, TRIMMED_LEGEND_WIDTH)
-      if (expandingLabelContainer >= maxLegendLabelWidth) {
+      if (expandingLabelContainer >= maxLegendLabelWidth || !trimLegend) {
```
Comment (Goondrious) [inline src/shared/utils/index.js:163]: nice

File: src/components/scatter-chart/events.js (hunk 2 above)
Comment (Goondrious) [reply src/components/scatter-chart/events.js:33]: (same for above)

File: src/components/scatter-chart/events.js (hunk 1 above)
Comment (geoerika) [reply src/components/scatter-chart/events.js:15]: yeah...I got lost looking deep for the structure of the node children....

**Pull request merged**